### PR TITLE
Low pressure environment Ripley buff & Mecha Mining Scanner.

### PR DESCRIPTION
**Pull request by xxalpha**
Changed Ripley movement so it moves faster in low pressure environments. The speed increases to almost the speed of a Gygax, but remains slower.

When in low pressure turfs, the Ripley will also drill twice as fast.

edit: Added a mech mining scanner, works automatically like the advanced mining scanner but without the ability to defuse gibtonite.

### CODE REVIEW

**Comment by WJohn** [comment]
I'm okay with these buffs, as it makes the ripley strictly only good while mining and against mining mobs, but not against the crew as a combat mech :+1:

**Comment by phil235** [comment]
:+1:

**Comment by xxalpha** [comment]
The sprite for the mech mining scanner is a recolor of the advanced mining scanner sprite (purple to orange). If someone wants to make a proper sprite, be my guest.

**Comment by GunHog** [comment]
:+1:

**Comment by optimumtact** [comment]
Time to de pressurise cargo

File: code/game/mecha/equipment/tools/tools.dm [hunk 1]
```diff
@@ -176,6 +176,48 @@
...
+	icon_state = "mecha_analyzer"
+	origin_tech = "materials=3;engineering=2"
+	equip_cooldown = 30
+	var/scanning = 0
+
+/obj/item/mecha_parts/mecha_equipment/tool/mining_scanner/New()
+	SSobj.processing |= src
+
+/obj/item/mecha_parts/mecha_equipment/tool/mining_scanner/process()
+	if(!loc)
```
Comment (Cheridan) [inline code/game/mecha/equipment/tools/tools.dm]: Assuming that this chunk is copy-pasted from the mining scanner code. Please make it into a proc if this is the case.

Comment (xxalpha) [inline code/game/mecha/equipment/tools/tools.dm]: Yes.

**Comment by Kromgar** [comment]
So uh... I fucking love you. xxalpha. Finally i don't need to get a advanced scanner before i mine

**Comment by xxalpha** [comment]
Made a proc. @Cheridan

**Pull request merged**